### PR TITLE
zlib: support building when conan is run in msys2 environment

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **zlib/1.2.11**

While triaging https://github.com/conan-io/conan-center-index/issues/3196, I could not get zlib to build under msys2.
This means running conan in a msys2 prompt with the msys2 gcc toolchain.
This should also fix https://github.com/conan-io/conan-center-index/issues/3521 (but that is about cygwin)

Tested on:
- Windows msvc
- Windows mingw
- Windows gcc@msys2

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by prince-chrismc** [comment]
I see the PR built but was marked as a draft, is there a reason it's not ready?

**Comment by madebr** [comment]
I created this pr while triaging https://github.com/conan-io/conan-center-index/issues/3196
I believe the method to detect msys2 is wrong, it should use `self.settings.os.subsystem`.

File: recipes/zlib/1.2.11/conanfile.py [hunk 1]
```diff
@@ -78,6 +78,7 @@ def _build_zlib_autotools(self):
 
     def _build_zlib_cmake(self):
         cmake = CMake(self)
+        cmake.verbose = True
```
Comment (Croydon) [inline recipes/zlib/1.2.11/conanfile.py:81]: Don't forget this one

Comment (madebr) [inline recipes/zlib/1.2.11/conanfile.py:81]:
```suggestion
```

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**